server/pbench/bin/pbench-move-unpacked

```diff
@@ -0,0 +1,181 @@
+#! /bin/bash
+# -*- mode: shell-script -*-
+
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+TMP_PATH=$1
+
+# TOP, ARCHIVE, INCOMING, RESULTS are all defined by the base file
+. $dir/pbench-base.sh
+. $dir/job_pool.sh
+###########################################################################
+
+log_init $(basename $0)
+
+# make sure only one copy is running.
+# Use 'flock -n $LOCKFILE /home/pbench/bin/pbench-move-unpacked' in the
+# crontab to ensure that only one copy is running. The script itself
+# does not use any locking.
+
+# the link source and destination for this script
+linksrc=TODO
+linkdest=TO-INDEX
+
+echo $TS
+
+# get the list of files we'll be operating on - sort them by size
+list=$TMP_PATH/pbench-unpack-tarballs.$$
+find -L $ARCHIVE/*/$linksrc -name '*.tar.xz' -printf "%s\t%p\n" 2>/dev/null | grep -v DUPLICATE | sort -n > $list
+
+trap 'rm -f $list' EXIT INT QUIT
+
+typeset -i ntb=0
+typeset -i ntotal=0
+typeset -i nerrs=0
+typeset -i ndups=0
+
+function process_tarballs {
+    result=$1
+    size=$2
+    ntotal=$ntotal+1
+
+    link=$(readlink -e $result)
+    if [ ! -f "$link" ] ;then
+        echo "$TS: $link does not exist" >&4
+        nerrs=$nerrs+1
+        continue
```

Comment:
There is no loop in this function to continue. Presumably you want to return. This applies to all the error handling in this function.